app/controllers/core_data_connector/names_controller.rb

```diff
@@ -0,0 +1,31 @@
+module CoreDataConnector
+  class NamesController < ApplicationController
+    # Search attributes
+    search_attributes :name
+
+    def base_query
+      query = super
+
+      return query.none unless params[:project_id] && params[:nameable_model]
+
+      nameable_table = params[:nameable_model].to_sym
+      nameable_model = "CoreDataConnector::#{params[:nameable_model].classify}".constantize
```

Comment:
So we're allow a user to select from a list of names for a particular model within a project. So if I have a `Work` record with a name "A Tale of Two Cities", I can only reuse that name on other `Work` records?

Reply:
The scoping is based on the `polymorphic_table` (a few lines down), in this case `source_titles`. Models that share a polymorphic M2M name relation table should be able to share their list of names. In other words, Instance, Item, and Work should be able to reuse names across all three models because they share the `source_titles` M2M table. Do you see anything here that would make it work differently?

Reply:
So presumably this controller would get a request as:
```
/core_data/names?project_id=1&nameable_model=work
```

Removing all of the meta-programming, the resulting query would look something like this:
```
query
  .where(
    SourceTitle
       .joins(works :project_model)
       .where(project_models: {
         project_id: params[:project_id]
       })
       .where(SourceTitle.arel_table[:nameable_type].eq(Work.to_s))
       .where(SourceTitle.arel_table[:name_id].eq(CoreDataConnector::Name.arel_table[:id]))
       .arel
       .exists
  )
```

Wouldn't the `where(SourceTitle.arel_table[:nameable_type].eq(Work.to_s))` filter this query to only include `source_titles` records that are linked to `works`?

This should also be easily testable by adding a name to one type of model and seeing if it is available when creating another model.